src/main/java/com/google/api/codegen/transformer/csharp/CSharpApiMethodTransformer.java

```diff
@@ -310,4 +270,80 @@ protected void setServiceResponseTypeName(
 
     return apiMethods;
   }
+
+  private List<StaticLangApiMethodView> generateFlattenedLroMethods(MethodContext methodContext) {
+    List<StaticLangApiMethodView> apiMethods = new ArrayList<>();
+    apiMethods.add(
+        generateAsyncOperationFlattenedMethod(
+            // TODO: replace the empty list with real calling forms:
+            //
+            // LongRunningFlattenedAsyncPollUntilComplete
+            // LongRunningFlattenedAsyncPollLater
+            //
+            // Keep an empty list here to turn off generating samples for
+            // LRO methods for now so that the baseline does not explode
+            methodContext.withCallingForms(Collections.emptyList()),
+            csharpCommonTransformer.callSettingsParam(),
+            ClientMethodType.AsyncOperationFlattenedCallSettingsMethod,
+            true));
+    apiMethods.add(
+        generateAsyncOperationFlattenedMethod(
+            methodContext.withCallingForms(Collections.emptyList()),
+            csharpCommonTransformer.cancellationTokenParam(),
+            ClientMethodType.AsyncOperationFlattenedCancellationMethod,
+            true));
+    apiMethods.add(
+        generateOperationFlattenedMethod(
+            // TODO: replace the empty list with real calling forms:
+            //
+            // LongRunningFlattenedPollUntilComplete
+            // LongRunningFlattenedPollLater
+            //
+            // Keep an empty list here to turn off generating samples for
+            // LRO methods for now so that the baseline does not explode
+            methodContext.withCallingForms(Collections.emptyList()),
+            csharpCommonTransformer.callSettingsParam()));
+    return apiMethods;
+  }
+
+  private List<StaticLangApiMethodView> generatePageStreamingFlattenedMethods(
+      MethodContext methodContext, List<ParamWithSimpleDoc> pagedMethodAdditionalParams) {
+    List<StaticLangApiMethodView> apiMethods = new ArrayList<>();
+    apiMethods.add(
+        generatePagedFlattenedAsyncMethod(
+            methodContext.withCallingForms(
+                ImmutableList.of(
+                    CallingForm.FlattenedAsyncPaged,
+                    CallingForm.FlattenedAsyncPagedAll,
+                    CallingForm.FlattenedAsyncPagedPageSize)),
+            pagedMethodAdditionalParams));
+    apiMethods.add(
+        generatePagedFlattenedMethod(
+            methodContext.withCallingForms(
+                ImmutableList.of(
+                    CallingForm.FlattenedPaged,
+                    CallingForm.FlattenedPagedAll,
+                    CallingForm.FlattenedPagedPageSize)),
+            pagedMethodAdditionalParams));
+    return apiMethods;
+  }
+
+  private List<StaticLangApiMethodView> generateFlattenedAsyncMethods(MethodContext methodContext) {
```

Comment:
Minor point, but this probably shouldn't be named `generateFlattenedAsyncMethods`, as it's generating a sync method as well as two async methods.
Possibly `generateNormalFlattenedMethods`?

Reply:
Ok!